bootstrap

```diff
@@ -1,25 +1,28 @@
 #!/bin/sh
+# Version: v.23.08.03
 
 set -e
 
 export LIBRARIES_REPOSITORIES='https://github.com/sunflowerit/waftlib.git'
-export LIBRARIES_VERSION_BRANCH=${LIBRARIES_VERSION_BRANCH:-v.22.05.30} #versions: v.21.05.10, v.21.09.22, v.22.05.30
+# Libraries version branches: v.21.05.10, v.21.09.22, v.22.05.30, v.23.08.03
+export LIBRARIES_VERSION_BRANCH="${LIBRARIES_VERSION_BRANCH:-v.23.08.03}"
 
 ################################################################
 ######                                                    ######
 ######       Don't change variables after this line       ######
 ######                                                    ######
 ################################################################
 
-SCRIPT_PATH=$(cd "$(/usr/bin/dirname "$0")" && /bin/pwd)
-ODOO_WORK_DIR=$(cd $SCRIPT_PATH && /bin/pwd)
-cd $ODOO_WORK_DIR
+SCRIPT_PATH="$(cd "$(/usr/bin/dirname "${0}")" && /bin/pwd)"
+ODOO_WORK_DIR="$(cd "${SCRIPT_PATH}" && /bin/pwd)"
+cd "${ODOO_WORK_DIR}"
 
-/bin/echo "INFO: Remove exist $ODOO_WORK_DIR/waftlib"
-/bin/rm -fr $ODOO_WORK_DIR/waftlib
-/bin/echo "INFO: Clone $LIBRARIES_REPOSITORIES to $ODOO_WORK_DIR/waftlib"
-/usr/bin/git clone -b $LIBRARIES_VERSION_BRANCH $LIBRARIES_REPOSITORIES $ODOO_WORK_DIR/waftlib 2> /dev/null || \
-/usr/bin/git clone -b master $LIBRARIES_REPOSITORIES $ODOO_WORK_DIR/waftlib
+/bin/echo "( INFO ) : Remove exist ${ODOO_WORK_DIR}/.ignore/waftlib"
+/bin/rm -fr "${ODOO_WORK_DIR}/.ignore/waftlib" && \
+/bin/mkdir -p "${ODOO_WORK_DIR}/.ignore/waftlib"
```

Comment:
I will need time for a full review, but one thing I have doubts about is the `.ignore` directory,  I understand that contains all the stuff that git should not add to the repository, but having a few more entries in .gitignore seems to me a very minor problem.

I think it might help in understanding the structure if we follow the conventions used in a lot of projects. Like putting libraries needed before anything in a lib directory (`./lib/waftlib`), putting every artefact created by build in a `./build` directory.

Maybe not crucial for how it all works, but when we are moving and reorganizing directories anyway, why not?

Reply:
>I will need time for a full review, 
take your time.
>but one thing I have doubts about is the .ignore directory, I understand that contains all the stuff that git should not add to the repository, but having a few more entries in .gitignore seems to me a very minor problem.
in my opinion, just the all stuff  in `.ignore` directory that git should not add to the repository, not `.ignore` directory contains all the stuff that git should not add to the repository.
>I think it might help in understanding the structure if we follow the conventions used in a lot of projects. Like putting libraries needed before anything in a lib directory (./lib/waftlib), putting every artefact created by build in a ./build directory.
in my opinion, we don't have any library in this code that we import from outside, this project is not a python code, it is a few scripts (sometimes python scripts, sometimes bash or shell scripts), and 'lib' directories confused others, we have a small code that we use it in many scripts so we putted it in `/.venv-*/lib/python3.10/site-packages/venv*lib/` as link, and I didn't see it is a reason to have a directory for it in the main project directory.

>Maybe not crucial for how it all works, but when we are moving and reorganizing directories anyway, why not?
both are good for me, just I did what I liked, we can discuss it more with other members in our team.